### PR TITLE
Resolve postgres double free

### PR DESCRIPTION
**Pull request by zaneselvans**
# PR Overview

Pin OpenSSL to avoid a bad interaction with PostgreSQL which was breaking the nightly builds.

# PR Checklist

- [ ] Merge the most recent version of the branch you are merging into (probably `dev`).
- [ ] All CI checks are passing. [Run tests locally to debug failures](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#running-tests-with-tox)
- [ ] Make sure you've included good docstrings.
- [ ] For major data coverage & analysis changes, [run data validation tests](https://catalystcoop-pudl.readthedocs.io/en/latest/dev/testing.html#data-validation)
- [ ] Include unit tests for new functions and classes.
- [ ] Defensive data quality/sanity checks in analyses & data processing functions.
- [ ] Update the [release notes](https://catalystcoop-pudl.readthedocs.io/en/latest/release_notes.html) and reference reference the PR and related issues.
- [ ] Do your own explanatory review of the PR to help the reviewer understand what's going on and identify issues preemptively.

### CODE REVIEW

File: environments/conda-lock.yml
Comment (zaneselvans) [inline environments/conda-lock.yml:1]: Why did basically every row of this file change?

Comment (bendnorman) [reply environments/conda-lock.yml:1]: That seems fishy

**Comment by zaneselvans** [comment]
And why the heck is `pip` unable to install PUDL on RTD?
```
INFO: pip is looking at multiple versions of catalystcoop-pudl to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement openssl!=3.2.0,>=3.1.4 (from catalystcoop-pudl) (from versions: none)
ERROR: No matching distribution found for openssl!=3.2.0,>=3.1.4
```

**Comment by zaneselvans** [comment]
Replaced by #3135

**Pull request closed without merging**